packages/build/tests/plugins/fixtures/config_readonly_define/plugin.js

```diff
@@ -2,6 +2,6 @@
 
 module.exports = {
   onPreBuild({ netlifyConfig }) {
-    Object.defineProperty(netlifyConfig.build, 'ignore', { value: '' })
+    Object.defineProperty(netlifyConfig.build, 'ignore', { value: '', enumerable: true })
```

Comment:
The new algorithm ignores non-enumerable properties changed/added/removed on `netlifyConfig`.